llvm/lib/TargetParser/RISCVISAInfo.cpp

```diff
@@ -492,7 +492,9 @@ RISCVISAInfo::parseNormalizedArchString(StringRef Arch) {
                                "'" + Twine(ExtName[0]) +
                                    "' must be followed by a letter");
 
-    ISAInfo->addExtension(ExtName, {MajorVersion, MinorVersion});
+    if (!ISAInfo->addExtension(ExtName, {MajorVersion, MinorVersion}))
+      return createStringError(errc::invalid_argument,
+                               "duplicate extension '" + ExtName + "'");
```

Comment:
Could we inform the user which version was ultimately chosen?

Reply:
> Could we inform the user which version was ultimately chosen?

We don't chose a version, we fail the parsing.

Reply:
Thanks for the explanation!